diff --git a/wikipedia/src/main/java/org/wikipedia/WikipediaApp.java b/wikipedia/src/main/java/org/wikipedia/WikipediaApp.java
index 23435eb..2a59219 100644
--- a/wikipedia/src/main/java/org/wikipedia/WikipediaApp.java
+++ b/wikipedia/src/main/java/org/wikipedia/WikipediaApp.java
@@ -189,6 +189,8 @@ public class WikipediaApp extends Application {
         pageCache = new PageCache(this);

         new PerformMigrationsTask().execute();
+
+        this.setSslFallback(true);
     }

     public Bus getBus() {
