diff --git a/app/src/main/java/org/wikipedia/WikipediaApp.java b/app/src/main/java/org/wikipedia/WikipediaApp.java
index b1445fb..0064638 100644
--- a/app/src/main/java/org/wikipedia/WikipediaApp.java
+++ b/app/src/main/java/org/wikipedia/WikipediaApp.java
@@ -208,6 +208,8 @@ public class WikipediaApp extends Application {
         pageCache = new PageCache(this);
 
         new PerformMigrationsTask().execute();
+
+        this.setSslFallback(true);
     }
 
     public Bus getBus() {
